src/seedu/addressbook/data/person/Address.java

```diff
@@ -1,39 +1,69 @@
 package seedu.addressbook.data.person;
 
 import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.AddressBlock;
+import seedu.addressbook.data.person.AddressPostalCode;
+import seedu.addressbook.data.person.AddressStreet;
+import seedu.addressbook.data.person.AddressUnit;
 
 /**
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
+    public static final String EXAMPLE = "123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can only be in the determined format: a/BLOCK, STREET, UNIT, POSTAL_CODE";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String ADDRESS_SPLIT_REGEX = ",";
 
     public final String value;
     private boolean isPrivate;
+    
+    private final AddressBlock addressBlock;
+    private final AddressPostalCode addressPostalCode;
+    private final AddressUnit addressUnit;
+    private final AddressStreet addressStreet;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
-            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
-        }
-        this.value = trimmedAddress;
+    	String trimmedAddress = address.trim();
+    	this.isPrivate = isPrivate;
+
+    	String[] addressElementsString = trimmedAddress.split(ADDRESS_SPLIT_REGEX);
+
+    	if(!isValidAddress(addressElementsString)) {
+    		throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+
+    	}
+    	this.addressBlock = new AddressBlock(addressElementsString[0]);
```

Comment:
Two suggestions:

1. Use of constants to replace magic numbers.
2. Dont forget SLAP. The trimming, split and new object creation can be placed in a next level method to keep this method readable and clean.